### PR TITLE
feat: support Source Map in TypeScript SDK

### PR DESCRIPTION
**Pull request by TomChv**
Depends on #9031

Implement `SourceMap` to display the actual position of the definition when codegen.

It's very useful to navigate inside the codebase when using local dependencies since generated function to call
the dependency doesn't directly use the dep's code but instead call it through the Dagger API.

Same as go in #8498

### TODO
- [x] Registration by the TS SDK (tested manually for now)
- [x] Generation on the TS client (tested manually for now)
- [x] Integration tests
- [x] Add column support
- [ ] Add doc

### Example of generation

```ts
export type TestLocationContainerOpts = {
  name: string // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:25)
}

export type TestLocationHumanOpts = {
  name: string // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:20)
  age: number // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:20)
}

export enum TestLocationFoo { // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:6)
  Bar = "bar",
  Baz = "baz",
  Foo = "foo",
}

export class TestLocationPerson extends BaseClient { // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:12)
  ...

  age = async (): Promise<number> => { // test-location (../../../../../../../dagger/location/ts-to-go/test-location/src/index.ts:12)
  ... 
  }
}
```

### Demo

https://github.com/user-attachments/assets/f64654b8-c3ff-4a6f-b127-29011b9993c8

### CODE REVIEW

**Comment by gerhard** [comment]
Looking good @TomChv - love the demo screen recording!

Please add a todo item to add **something** (anything) to the docs. I know that @vikram-dagger will enjoy pairing-up with you on this.

**Review by jedevc — COMMENTED** [review]
Nice - this looks pretty good to me on first glance, looking forward to the tests!

Just a little note, potentially you might want to update https://github.com/dagger/dagger/blob/3cc7cfb93d097b83e3c9cc585d7f6bf35f125bfa/cmd/codegen/introspection/introspection.go#L297-L299

We could put a column here, as discussed separately - not every editor understands it, but it is useful metadata we can include!

File: sdk/typescript/src/module/introspector/typescript_module/ast.ts [hunk 1]
```diff
@@ -108,6 +110,43 @@ export class AST {
...
+
+    // Use the identifier of the node if available.
+    const targetNode = node.name ?? node
+
+    const position = ts.getLineAndCharacterOfPosition(
+      sourceFile,
+      targetNode.getStart(sourceFile),
+    )
+
+    // sourcile.filaname is the absolute path to the file, we need to get the relative path
```
Comment (helderco) [inline sdk/typescript/src/module/introspector/typescript_module/ast.ts:136]:
```suggestion
    // sourceFile.fileName is the absolute path to the file, we need to get the relative path
```

**Review by helderco — APPROVED** [review]
🚀

**Pull request merged**